### ISSUE TITLE
Received SMS sensor

### ISSUE THREAD

**Issue by thedexboy** [home-assistant/android#1699]
**Is your feature request related to a problem? Please describe.**

I was wondering if there was an option for HA to pick up the last received SMS and expose it as a text sensor so it could be used for automations? One example is that I sometimes miss the garbage truck because the SMS got hidden in all the notifications. I could make an automation to notify me to get the trash cans ready until I confirm I've done it (SMS arrives a day early at around 6 PM, collection is usually in the morning).

**Describe the solution you'd like**

Ability to expose the last received SMS message as a sensor. Well, 2 sensors: the originating phone number, second for the message itself.
(optionally 3rd sensor would be on what SIM card the message arrived, but just getting the SMS forwarded would be great!)

**Describe alternatives you've considered, if any**

I'm currently using Tasker and Join (paid version) to send the last received SMS to Node-Red but it's so unreliable that it's nearly useless. I was also pushing notifications to my phone but it was very unreliable, mostly when phone was switching from home Wi-Fi to 4G. HA app doesn't have this problem, message may be delayed but it will arrive and this is a big plus :)

**Additional context**

If it helps I can beta-test things but I can't help with coding unfortunately.

Thanks!

**Comment by dshokouhi** [comment]
Last Notification Sensor is what you want, all details from the notification will be sent to HA. You can also set an allow list to only get notifications from a particular app. More details here.

https://companion.home-assistant.io/docs/core/sensors#last-notification

**Comment by thedexboy** [comment]
Thanks for the idea!
I tested it and got mixed results. What happens is that if I get 2+ messages at pretty much the same time (one for mixed trash, other for biodegradable waste) only the last one is shown.

Another thing that I tested: if my phone isn't on a good network connection when I get the SMS, it will never be uploaded to HA because another notification will become the last one.

I couldn't figure out any workarounds for this :(
